File: Problem-13/Multiplication_2D.py
# WAP to find a Multiplication of 2 Matrix.

r1 = int(input("Enter rows of matrix 1 : "))
c1 = int(input("Enter columns of matrix 1 : "))


r2 = int(input("Enter rows of matrix 2 : "))
c2 = int(input("Enter columns of matrix 2 : "))


if c1 != r2 :
    print("Matrix multiplication not possible !!")
    exit()


print("Enter elements of Matrix 1 : ")
arr1 = []
for i in range(r1) :
    row = []
    for j in range(c1) :
        element = int(input(f"arr1[{i+1}][{j+1}] : "))
        row.append(element)
    arr1.append(row)


print("Enter elements of Matrix 2 : ")
arr2 = []
for i in range(r2):
    row = []
    for j in range(c2) :
        element = int(input(f"arr2[{i+1}][{j+1}] : "))
        row.append(element)
    arr2.append(row)


ans = [[0 for _ in range(c2)] for _ in range(r1)]


for i in range(r1) :
    for j in range(c2) :
        for k in range(c1) :
            ans[i][j] += arr1[i][k] * arr2[k][j]


print("\nMatrix 1 : ")
for i in arr1 :
    print(i)

print("\nMatrix 2 : ")
for i in arr2 :
    print(i)

print("\nMultiplication Result : ")
for i in ans :
    print(i)

File: Problem-13/Multiplication_2D.c
// WAP to find the Multiplication of 2 Matrices
#include <stdio.h>

int main() {
    int r1, c1, r2, c2;

    printf("Enter rows of matrix 1 : ");
    scanf("%d", &r1);
    printf("Enter columns of matrix 1 : ");
    scanf("%d", &c1);

    printf("Enter rows of matrix 2 : ");
    scanf("%d", &r2);
    printf("Enter columns of matrix 2 : ");
    scanf("%d", &c2);

    if (c1 != r2) {
        printf("Matrix multiplication not possible !!\n");
        return 0;
    }

    int arr1[r1][c1], arr2[r2][c2], ans[r1][c2];

    printf("Enter elements into arr1 :\n");
    for (int i = 0; i < r1; i++) {
        for (int j = 0; j < c1; j++) {
            printf("arr1[%d][%d] : ", i, j);
            scanf("%d", &arr1[i][j]);
        }
    }

    printf("Enter elements into arr2 :\n");
    for (int i = 0; i < r2; i++) {
        for (int j = 0; j < c2; j++) {
            printf("arr2[%d][%d] : ", i, j);
            scanf("%d", &arr2[i][j]);
        }
    }

    for (int i = 0; i < r1; i++) {
        for (int j = 0; j < c2; j++) {
            ans[i][j] = 0;
            for (int k = 0; k < c1; k++) {
                ans[i][j] += arr1[i][k] * arr2[k][j];
            }
        }
    }

    printf("\nMatrix 1 :\n");
    for (int i = 0; i < r1; i++) {
        for (int j = 0; j < c1; j++) {
            printf("%d ", arr1[i][j]);
        }
        printf("\n");
    }

    printf("\nMatrix 2 :\n");
    for (int i = 0; i < r2; i++) {
        for (int j = 0; j < c2; j++) {
            printf("%d ", arr2[i][j]);
        }
        printf("\n");
    }

    printf("\nMultiplication Of Matrix :\n");
    for (int i = 0; i < r1; i++) {
        for (int j = 0; j < c2; j++) {
            printf("%d ", ans[i][j]);
        }
        printf("\n");
    }

    return 0;
}